### PR TITLE
Fix get_ppl & get_logits

### PR DESCRIPTION
**Pull request by lvhan028**
depends on 

- [x] #2968

### CODE REVIEW

File: lmdeploy/serve/utils.py [hunk 1]
```diff
@@ -188,21 +73,18 @@ def get_ppl(self, input_ids: Union[List[int],
...
-        generator = self.engine.create_instance()
-
         # TODO: a better way to determine `max_input_len`, at most allocate
         # 2G mem for logits with shape [bs, max_input_len, vocab_size]
         vocab_size = self.hf_tm_cfg.vocab_size
         max_input_len = 2 * 1024**3 // (vocab_size * 4)
         sizes = [len(_) for _ in input_ids]
-        losses = []
-        target_counts = []
+        result = []
```
Comment (irexyc) [inline lmdeploy/serve/utils.py:87]: should we check if all size in sizes > 1

File: lmdeploy/serve/utils.py [hunk 2]
```diff
@@ -16,166 +16,51 @@
...
+                                                output_logits='all'),
+                    stream_response=False,
+                    sequence_start=sequence_start,
+                    sequence_end=sequence_end):
+                # In the last iteration, the yielded `out` is an empty response
+                # indicating the finish_reason, which should be ignored here
+                if out.finish_reason is None:
+                    # Try not to return in async for loop. Otherwise, there
+                    # will be `GeneratorExit` exception
+                    logits[i] = out.logits
```
Comment (irexyc) [inline lmdeploy/serve/utils.py:59]: 这里的的 session_id 每次都递增，感觉 sequence_start, sequence_end, step 对应不上历史的 session，_get_long_text_ppl 可能有问题。

另外就是有一个场景，用户那边想要 logits，但是每次只会传 session 的一部分，由于上面的问题，目前的接口感觉实现不了。

Comment (lvhan028) [reply lmdeploy/serve/utils.py:59]: 获取 logits 的方式，在文档中说明了。不用之前的方法了。

get_ppl 中 session_id 的问题，我修一下。

File: lmdeploy/serve/utils.py (hunk 1 above)
Comment (irexyc) [reply lmdeploy/serve/utils.py:81]: I think `len(_)` should > 1 otherwise we will get nan value

**Review by irexyc — APPROVED** [review]

**Review by lzhangzz — APPROVED** [review]
